### PR TITLE
fix: if audience is filtering and no audiences exist on environments, add all environments

### PR DESCRIPTION
**Pull request by RohinBhargava**
added hack tests and overrideChecks -- have to do here in order to make sure that errant audiences don't pass

### CODE REVIEW

File: packages/cli/generation/ir-generator/src/__test__/irs/environmentAudiencesAllHack.json [hunk 1]
```diff
@@ -0,0 +1,4659 @@
...
+                                                "unsafeName": "id",
+                                                "safeName": "id"
+                                            },
+                                            "snakeCase": {
+                                                "unsafeName": "id",
+                                                "safeName": "id"
+                                            },
+                                            "screamingSnakeCase": {
+                                                "unsafeName": "ID",
+
```
Comment (RohinBhargava) [inline packages/cli/generation/ir-generator/src/__test__/irs/environmentAudiencesAllHack.json:4197]: check here

File: packages/cli/generation/ir-generator/src/__test__/irs/environmentAudiencesSelectHack.json [hunk 2]
```diff
@@ -0,0 +1,1529 @@
...
+            "wireValue": "errorInstanceId"
+        }
+    },
+    "environments": {
+        "defaultEnvironment": null,
+        "environments": {
+            "type": "singleBaseUrl",
+            "environments": [
+                {
+                    "id": "Production",
```
Comment (RohinBhargava) [inline packages/cli/generation/ir-generator/src/__test__/irs/environmentAudiencesSelectHack.json:1173]: and here

**Comment by RohinBhargava** [comment]
Old tests have no diff. Random audiences are not actually allowed in this layer

File: packages/cli/generation/ir-generator/src/generateIntermediateRepresentation.ts [hunk 3]
```diff
@@ -398,14 +398,45 @@ export async function generateIntermediateRepresentation({
...
+    const workspaceDefinitionRootApiFileContents = workspace.definition.rootApiFile.contents;
     const environments = convertEnvironments({
         casingsGenerator,
-        rawApiFileSchema: workspace.definition.rootApiFile.contents
+        rawApiFileSchema: workspaceDefinitionRootApiFileContents
     });
+
+    // TODO: (rohin) Back compat hack before pushing generator/api upgrade
+    let overrideCheck: boolean;
+    environments?.environments._visit({
```
Comment (dsinghvi) [inline packages/cli/generation/ir-generator/src/generateIntermediateRepresentation.ts:409]: `const overrideCheck = environments?.environments._visit<boolean>({...`

File: packages/cli/generation/ir-generator/src/filtered-ir/IrGraph.ts [hunk 4]
```diff
@@ -107,10 +107,12 @@ export class IrGraph {
 
     public markEnvironmentForAudiences(
         environment: SingleBaseUrlEnvironment | MultipleBaseUrlsEnvironment,
-        audiences: string[]
+        audiences: string[],
+        // TODO: (rohin) this is brought in for backwards compatibility when no audiences are supplied to environments. Restore this when api/generators bumping.
+        overrideCheck?: boolean
```
Comment (dsinghvi) [inline packages/cli/generation/ir-generator/src/filtered-ir/IrGraph.ts:112]: rename to `ignoreAudiences`

**Review by dsinghvi — APPROVED** [review]

**Pull request merged**